src/sentry/static/sentry/app/views/organizationEventsV2/tags.jsx

```diff
@@ -28,19 +27,22 @@ class Tags extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (
-      this.props.view.id !== prevProps.view.id ||
-      !isEqual(this.props.selection, prevProps.selection)
-    ) {
+    // Do not update if we are just opening/closing the modal
+    const locationHasChanged = !isEqual(
+      omit(prevProps.location.query, MODAL_QUERY_KEYS),
+      omit(this.props.location.query, MODAL_QUERY_KEYS)
+    );
+
+    if (this.props.view.id !== prevProps.view.id || locationHasChanged) {
       this.fetchData();
     }
   }
 
   fetchData = async () => {
-    const {api, organization, view} = this.props;
+    const {api, organization, view, location} = this.props;
 
     try {
-      const tags = await fetchTags(api, organization.slug, view.tags);
+      const tags = await fetchTags(api, organization.slug, view.tags, location.query);
```

Comment:
Does the fetchTags() action need to be updated to handle the datetime properties being structured differently?

Reply:
How are they structured differently?

Reply:
The data in the selection is nested inside a `datetime` property, whereas in the query string the  structure is flat.

Reply:
Ah, yeah it's just pulling it out of the browser's URL bar which is flat and not going through the stored structure